File: examples/governance_static_go/compliance_static_go/domain/compliance_review_repository.go
package domain

type ComplianceReviewRepository interface {
	Find(id string) (*ComplianceReview, error)
	Save(ComplianceReview *ComplianceReview) error
	All() ([]*ComplianceReview, error)
	Delete(id string) error
	Count() (int, error)
}

File: examples/governance/compliance_domain/lib/compliance_domain/ports/compliance_review_repository.rb
module ComplianceDomain
  module Ports
    module ComplianceReviewRepository
      def find(id)
        raise NotImplementedError, "#{self.class}#find not implemented"
      end

      def save(compliance_review)
        raise NotImplementedError, "#{self.class}#save not implemented"
      end

      def delete(id)
        raise NotImplementedError, "#{self.class}#delete not implemented"
      end
    end
  end
end
